modules/caddyhttp/templates/tplcontext.go

```diff
@@ -164,6 +166,18 @@ func (c templateContext) executeTemplateInBuffer(tplName string, buf *bytes.Buff
 	return parsedTpl.Execute(buf, c)
 }
 
+func (c templateContext) placeholder(name string) string {
+	repl := c.Req.Context().Value(caddy.ReplacerCtxKey).(*caddy.Replacer)
+
+	value, has := repl.Get(name)
+
+	if !has {
+		return ""
+	}
+
+	return fmt.Sprintf("%v", value)
+}
```

Comment:
```suggestion
	repl := c.Req.Context().Value(caddy.ReplacerCtxKey).(*caddy.Replacer)
	value, _ := repl.GetString(name)
	return value
}
```

Might need to rebase with master -- I know this GetString() method was added recently but I can't recall when, maybe it's still in a PR... in any case, please try this out.